src/io/leopard/toLeopard.ts

```diff
@@ -766,16 +766,29 @@ export default function toLeopard(
           return `${selectedWatcherSource}.visible = true`;
         case OpCode.data_hidelist:
           return `${selectedWatcherSource}.visible = false`;
-        case OpCode.procedures_call:
-          return `yield* this.${
-            // Get name of custom block script with given PROCCODE:
-            target.scripts.find(
-              script =>
-                script.hat !== null &&
-                script.hat.opcode === OpCode.procedures_definition &&
-                script.hat.inputs.PROCCODE.value === block.inputs.PROCCODE.value
-            ).name
-          }(${block.inputs.INPUTS.value.map(inputToJS).join(", ")})`;
+        case OpCode.procedures_call: {
+          // Get name of custom block script with given PROCCODE:
+          const procName = target.scripts.find(
+            script =>
+              script.hat !== null &&
+              script.hat.opcode === OpCode.procedures_definition &&
+              script.hat.inputs.PROCCODE.value === block.inputs.PROCCODE.value
+          ).name;
+
+          const procCall = `this.${procName}(${block.inputs.INPUTS.value.map(inputToJS).join(", ")})`;
+
+          // Warp-mode procedures execute all child procedures in warp mode as well. This means running the generator
+          // functions to completion.
+          if (warp) {
+            return `{
+              const ${procName}_result = ${procCall};
```

Comment:
Because generators are stateful, we execute this once to instantiate it, then call `next` on the instance. We must nest this inside a block to avoid this variable name being redeclared in case the procedure is called multiple times from within the same function.

Reply:
In fact, I'll just add this in as a comment